### DISCUSSION TITLE
React - Search Directory - Additional options like: Adding more filtering or sorting capabilities

### DISCUSSION THREAD

**Discussion by Trevino-Valerie** [pnp/sp-dev-fx-webparts#2016] [Ideas]
We have added the react - search directory component into our site, but I am searching for ways to add additional features/options like: 

- a feature to have the option to point to M365 or Azure AD or Both
- add more than one filter/sort
- Is there a way to filter inactive employees
- A way to assign certain tags to the job titles, like we do for our other search functions? MarComm, MarCom, Marketing and Communications, MarCom Administration, etc

**Comment by mariaarcena** [comment]
Hello everyone and/or anyone! To break down the list above:

1. a feature to have the option to point to M365 or Azure AD or Both?
2. add more than one filter/sort?
3. is there a way to filter inactive employees?
4. a way to assign certain tags to the job titles, like we do for our other search functions?

If we can concentrate requests 1 and 2, will someone be able to take a look at this? 😊 I also understand the authors and contributors can be quite busy as well so any tips and pointers where we should start looking into for requests 1 and 2 will be appreciated!

Like for 1, is there a specific thing we should be googling for?(I looked into the source code and googled a couple things and no luck so far 😔) Same thing for 2!

Hope any of the authors and/or contributors as well as the lurkers can help or give advice! ✨

**Reply by gretchunkim** [reply]
Instead of building on top of the People Directory react SPFX consider utilizing PnP Search (V4) since the PnP Search SPFX comes with filter already set in place.  I'll think adding the index react component into PnP Search (V4) is easier than to take a portion of the filter capacity into People Directory react SPFX. 

It could also be vice versa. Utilize the PnP Search (V4) and take some of these filter components into People Directory.

**Comment by netmatrixtech** [comment]
Please see that I've created my own enhancements to react-directory to facilitate filtering of results based on Entra ID Group Object ID. It'll be up to the contributors for this webpart to decide if they wish to add it into their project.
It should cater to most of your requests if you use an Entra Dynamic Group to apply the accounts you want shown in the results of the webpart.

[https://github.com/pnp/sp-dev-fx-webparts/discussions/6463](url)
cheers
Tim
